### PR TITLE
Chapter 11

### PR DESCRIPTION
**Pull request by wwwvikas123**
@Cakemanator and @yegorgeorgiyev, pls, check my home work out. Thanks.
Лосева В

### CODE REVIEW

File: Hadoop/manual.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- Docker-compose (2 and later);
+- Docker;
+
+!!! ALL COMMANDS must be executed from context where docker-compose files are situated !!!
+
+
+## To build images locally and start, run theese commands
+
+```
+docker-compose -f docker-compose_build.yaml && docker-compose -f docker-compose_build.yaml up -d
```
Comment (Cakemanator) [inline Hadoop/manual.md]: It is not building
image

File: Hadoop/worker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     wget https://gist.githubusercontent.com/rdaadr/2bedf24fd2721bad276e416b57d63e38/raw/640ee95adafa31a70869b54767104b826964af48/hdfs-site.xml && \
+     wget https://gist.githubusercontent.com/rdaadr/64b9abd1700e15f04147ea48bc72b3c7/raw/2d416bf137cba81b107508153621ee548e2c877d/core-site.xml && \
+     wget https://gist.githubusercontent.com/rdaadr/2f42f248f02aeda18105805493bb0e9b/raw/6303e424373b3459bcf3720b253c01373666fe7c/hadoop-env.sh && \
+     wget https://gist.githubusercontent.com/Stupnikov-NA/ba87c0072cd51aa85c9ee6334cc99158/raw/bda0f760878d97213196d634be9b53a089e796ea/yarn-site.xml && \
+     cp core-site.xml hadoop-env.sh hdfs-site.xml yarn-site.xml $HADOOP_HOME/etc/hadoop/ && \  
+install -d --group=$USER --owner=$USER /opt/mount1/datanode-dir && \
+install -d --group=$USER --owner=$USER /opt/mount2/datanode-dir && \
+install -d --group=$USER --owner=$USER /opt/mount1/nodemanager-local-dir && \
+install -d --group=$USER --owner=$USER /opt/mount2/nodemanager-log-dir && \
+install -d --group=$USER --owner=$USER /opt/mount1/nodemanagerlog-dir && \
```
Comment (Cakemanator) [inline Hadoop/worker/Dockerfile]: Something gone wrong
image

File: Hadoop/manual.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+### to stop them
+
+```
+docker-compose -f docker-compose_build.yaml down
+```
+
+## To download images and start containers, run this command
+
+```
+docker-compose up -d
```
Comment (Cakemanator) [inline Hadoop/manual.md:28]: Shal we add instructions on how to make it reachable outside of the virtual machine?
image

**Review by Cakemanator — CHANGES_REQUESTED** [review]

File: Hadoop/manual.md (hunk 1 above)
Comment (wwwvikas123) [reply Hadoop/manual.md]: my misstake. I've corrected command.

File: Hadoop/manual.md (hunk 3 above)
Comment (wwwvikas123) [reply Hadoop/manual.md:28]: becouse of security, it is better to make it reachable via proxy (exmpl, nginx)
for exmpl

        location / {
                proxy_pass http://127.0.0.1:9870/;
        }

File: Hadoop/worker/Dockerfile (hunk 2 above)
Comment (Cakemanator) [reply Hadoop/worker/Dockerfile]: Good practice: Conversations should be resolved by the person, who started it :)

File: Hadoop/manual.md (hunk 3 above)
Comment (Cakemanator) [reply Hadoop/manual.md:28]: Ok, but we have a "dumb" user who doesn't know how to do that. Can we do something to make it work "out of the box"?

Comment (wwwvikas123) [reply Hadoop/manual.md:28]: just to change docker-compose.yml like that
    ports:
      - 1000:8088
      - 9870:9870
and service vm1 will listen to 0.0.0.0
 **(for prod)** Nevertheless, as i said, it is better to open only 80 and 443 and make proxy pass to local interface  , or allow access only for ip from white list

File: Hadoop/worker/Dockerfile (hunk 2 above)
Comment (wwwvikas123) [reply Hadoop/worker/Dockerfile]: исправила образ и запушила в docker hub, перед запуском надо будет сделать docker-compose pull

File: Hadoop/manual.md (hunk 3 above)
Comment (Cakemanator) [reply Hadoop/manual.md:28]: Seems legit)
Approved

**Review by Cakemanator — APPROVED** [review]

**Pull request merged**